node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -45,30 +44,117 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
+        try {
+          InetAddress
+            .getAllByName(seed)
+        } catch {
+          case _: UnknownHostException =>
+            logger.debug(s"DNS seed $seed is unavailable")
+            Vector()
+        }
+      })
+      .distinct
+      .filter(_.isReachable(500))
+      .map(_.getHostAddress)
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers.toVector
+  }
 
-  def setPeerServices(
-      peer: Peer,
-      serviceIdentifier: ServiceIdentifier): Unit = {
-    _peerServices.put(peer, serviceIdentifier)
-    ()
+  /** Returns peers from hardcoded addresses taken from https://github.com/bitcoin/bitcoin/blob/master/contrib/seeds/nodes_main.txt */
+  def getPeersFromResources: Vector[Peer] = {
+    val source = Source.fromURL(getClass.getResource("/hardcoded-peers.txt"))
+    val addresses = source
+      .getLines()
+      .toVector
+      .filter(nodeAppConfig.torConf.enabled || !_.contains(".onion"))
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers
+  }
+
+  /** Returns all peers stored in db */
+  def getPeersFromDb: Future[Vector[Peer]] = {
+    val addressesF: Future[Vector[PeerDb]] =
+      PeerDAO().findAll()
+    val peersF = addressesF.map { addresses =>
+      val filteredAddresses = addresses.filter(
+        nodeAppConfig.torConf.enabled || _.networkId != AddrV2Message.TOR_V3_NETWORK_BYTE)
+      val inetSockets = filteredAddresses.map(a => {
+        NetworkUtil.parseInetSocketAddress(a.address, a.port)
+      })
+      val peers =
+        inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+      peers
+    }
+    peersF
+  }
+
+  def getPeersFromConfig: Vector[Peer] = {
+    val addresses = nodeAppConfig.peers.filter(
+      nodeAppConfig.torConf.enabled || !_.contains(".onion"))
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers
+  }
+
+  /** Returns peers randomly taken from config, db, hardcoded peers, dns seeds in that order */
+  def getPeers: Future[Vector[Peer]] = {
+    //currently this would only give the first peer from config
+    val peersFromConfig = getPeersFromConfig
+    lazy val peersFromDbF = getPeersFromDb
+    lazy val peersFromDnsSeeds = getPeersFromDnsSeeds
+    lazy val peersFromResources = getPeersFromResources
+    val maxConnectedPeers = 1
+
+    val allF = for {
+      peersFromDb <- peersFromDbF
+    } yield {
+      val ret = Vector.newBuilder[Peer]
+      ret ++= Random.shuffle(peersFromConfig).take(maxConnectedPeers)
+      if (maxConnectedPeers - ret.result().size > 0)
```

Comment:
I don't think it is safe to call `.result()` and then keep adding new elements possibly to the `VectorBuilder`. Here is the scaladoc from `.result()`

```
Produces a collection from the added elements.
After a call to result, the behavior of all other methods is undefined save for clear(). If clear() is called, then the builder is reset and may be used to build another instance.
Returns:
a collection containing the elements added to this builder.
```

So if i read that correctly, it seems that the behavior is undefined after the first call to `result()`